showcase/src/app/app.routes.ts

```diff
@@ -1,58 +1,33 @@
-	DemoKeyboardDirectivesComponent,
-	DemoLanguageSelectorComponent,
-	DemoLogoutComponent,
-	DemoMenuComponent,
-	DemoPaginationComponent,
-	DemoPrettyPrintComponent,
-	DemoSidebarComponent,
-	DemoSliderComponent,
-	DemoTableComponent,
-	DemoToastComponent,
-	DemoTypographyComponent
-} from "./demo";
 import { HomeComponent } from "./home";
 import { NoContentComponent } from "./no-content";
 import { Ng2StateDeclaration } from "@uirouter/angular";
-import { NewsComponent } from "./news";
+import { AppComponent } from "./app.component";
 
 export const APP_STATES: Ng2StateDeclaration[] = [
-	{ name: "news", url: "/news", component: NewsComponent },
-	{ name: "home", url: "/", component: HomeComponent },
-	{ name: "demo-action-bar", url: "/demo/action-bar", component: DemoActionBarComponent },
-	{ name: "demo-breadcrumb", url: "/demo/breadcrumb", component: DemoBreadcrumbComponent },
-	{ name: "demo-button", url: "/demo/button", component: DemoButtonComponent },
-	{ name: "demo-card", url: "/demo/card", component: DemoCardComponent },
-	{ name: "demo-collapsible", url: "/demo/collapsible", component: DemoCollapsibleComponent },
-	{ name: "demo-colors", url: "/demo/colors", component: DemoColorsComponent },
-	{ name: "demo-date-picker", url: "/demo/date-picker", component: DemoDatePickerComponent },
-	{ name: "demo-date-range-picker", url: "/demo/date-range-picker", component: DemoDateRangePickerComponent },
-	{ name: "demo-dropdown", url: "/demo/dropdown", component: DemoDropdownComponent },
-	{ name: "stark-footer", url: "/demo/stark-footer", component: DemoFooterComponent },
-	{ name: "stark-header", url: "/demo/stark-header", component: DemoHeaderComponent },
-	{ name: "demo-example-viewer", url: "/demo/example-viewer", component: DemoExampleViewerComponent },
-	{ name: "demo-keyboard-directives", url: "/demo/keyboard-directives", component: DemoKeyboardDirectivesComponent },
-	{ name: "demo-language-selector", url: "/demo/language-selector", component: DemoLanguageSelectorComponent },
-	{ name: "demo-logout", url: "/demo/logout", component: DemoLogoutComponent },
-	{ name: "demo-menu", url: "/demo/menu", component: DemoMenuComponent },
-	{ name: "demo-pagination", url: "/demo/pagination", component: DemoPaginationComponent },
-	{ name: "demo-pretty-print", url: "/demo/pretty-print", component: DemoPrettyPrintComponent },
-	{ name: "demo-slider", url: "/demo/slider", component: DemoSliderComponent },
-	{ name: "demo-sidebar", url: "/demo/sidebar", component: DemoSidebarComponent },
-	{ name: "demo-table", url: "/demo/table", component: DemoTableComponent },
-	{ name: "demo-toast", url: "/demo/toast", component: DemoToastComponent },
-	{ name: "demo-typography", url: "/demo/typography", component: DemoTypographyComponent },
-	{ name: "otherwise", url: "/otherwise", component: NoContentComponent }
+	{
+		name: "app",
+		url: "/",
+		component: AppComponent
+	},
+	{
+		name: "home",
+		url: "",
+		views: { "@": { component: HomeComponent } },
```

Comment:
Why having the views: `{"@": {}}` here ?

Reply:
Well, because now that we implemented lazy loaded modules, the states graph was not correct and any navigation to lazy loaded modules failed.

So, I had to adapt the graph: basically I added the AppComponent as the root state of the app, so that means that the `<ui-view>` is no longer in the root Ui Router state but in a child route, therefore the `views: { "@": { component: xxxxx } }` is needed to render the page in that `ui-view`.

![showcase router state graph](https://user-images.githubusercontent.com/5698318/47799759-3528d200-dd2b-11e8-8029-f415513e1d13.PNG)